pkg/pillar/cmd/zedagent/attesttask.go

```diff
@@ -244,14 +246,19 @@ func (server *VerifierImpl) SendAttestQuote(ctx *zattest.Context) error {
 		return zattest.ErrControllerReqFailed
 	}
 
+	attestReq.Quote = quote
+
 	if attestCtx.EventLogParseErr == nil {
 		//On some platforms, either the kernel does not export TPM Eventlog
 		//or the TPM does not have SHA256 bank enabled for PCRs. We populate
 		//eventlog if we are able to parse eventlog successfully
-		encodeEventLog(attestCtx, quote)
-	}
+		encodeEventLog(attestCtx, attestReq.Quote)
 
-	attestReq.Quote = quote
+		if len(attestReq.Quote.EventLog) > 0 && proto.Size(attestReq) > maxQuotePayloadSize {
+			log.Errorf("[ATTEST] attestReq size too much (%d) will remove large binaries", proto.Size(attestReq))
+			cleanupEventLog(attestReq.Quote)
+		}
```

Comment:
If after the cleanup it is still too bit it would make sense to generate a log.Warnf